repository_service_tuf_api/__init__.py

```diff
@@ -194,20 +202,40 @@ def release_bootstrap_lock():
     redis_loader.write(settings_repository, settings_data)
 
 
-def is_bootstrap_done():
+def bootstrap_state() -> BootstrapState:
     """
-    Check if the boot is done.
+    Bootstrap state
+
+    The bootstrap state is registered in Redis.
+    Detailed definitions are available in
+    https://repository-service-tuf.readthedocs.io/en/stable/devel/design.html#tuf-repository-settings  # noqa
     """
+
     # Reload the settings
     # The reload is required because the settings object is created in the
     # `app.py`'s initialization. The `settings_repository.get_fresh() doesn't
     # correctly reload it because that and the settings_repository.reload() do
     # the job.
     settings_repository.reload()
-    if settings_repository.get_fresh("BOOTSTRAP", None) is None:
-        return False
-    else:
-        return True
+    bootstrap = settings_repository.get_fresh("BOOTSTRAP")
+
+    bootstrap_state = BootstrapState(bootstrap=False, state=None, task_id=None)
+    if bootstrap is None:
+        return bootstrap_state
+
+    if len(bootstrap.split("-")) == 1:
+        bootstrap_state.bootstrap = True
+        bootstrap_state.state = "finished"
+        bootstrap_state.task_id = bootstrap
+
+        return bootstrap_state
+
+    elif len(bootstrap.split("-")) == 2:
```

Comment:
I think it will be easier to understand if you check if the `bootstrap` value starts with `pre`.
I just realized it's possible that a real task id could start with `pre` prefix.
Then I see two possibilities:
1. instead of using `pre-<task_id>` as an intermediate state why don't you use capital letters e.g. `PRE-<task_id>`? 
I see we don't use capital letters for task ids.
2. If you don't like the first solution, then at least leave a comment on what will be the structure of the `bootstrap` value.

Reply:
I don't think you can have "pre-"
We use the https://github.com/repository-service-tuf/repository-service-tuf-api/blob/129f4dcadbc0bb0f2e4ba63f0112bd42172e3c8c/repository_service_tuf_api/__init__.py#L213-L214 which gives us a hexadecimal string without dashes (`-`).

Reply:
Okay, you are right, then check for `pre-` or `signing-`. Those were the possible intermediate values, right?

Reply:
Any intermediate `<state>-` is a `bootstrap=False`
We consider only `bootstrap=True`, when doesn't have intermediate states.
The intermediate states can be used by specific checks (at the moment, no use case but with DAS we will have one)

Reply:
Yes, that's my point. To make it clear.
If you don't want to make a difference to the check, then leave a comment that in this situation the bootstrap is at an intermediate state and that all intermediate state are in the form of `<STATE>-<task_id>.